fhd_core/calibration/vis_calibrate_crosspol_phase.pro

```diff
@@ -13,24 +13,33 @@ n_baselines=obs.nbaselines
 weights_use = 0>Reform(*vis_weight_ptr[0],n_freq,n_baselines,n_time)<1
 
 ;average the visibilities in time
-vis_xy = Reform(*vis_ptr[3],n_freq,n_baselines,n_time)
+vis_xy = Reform(*vis_ptr[2],n_freq,n_baselines,n_time)
 vis_xy = Total(Temporary(vis_xy)*weights_use,3)
-vis_yx = Reform(*vis_ptr[2],n_freq,n_baselines,n_time)
+vis_yx = Reform(*vis_ptr[3],n_freq,n_baselines,n_time)
 vis_yx = Total(Temporary(vis_yx)*weights_use,3)
+model_xy = Reform(*vis_model_ptr[2],n_freq,n_baselines,n_time)
+model_xy = Total(Temporary(model_xy)*weights_use,3)
+model_yx = Reform(*vis_model_ptr[3],n_freq,n_baselines,n_time)
+model_yx = Total(Temporary(model_yx)*weights_use,3)
 
 ;remove zeros
 weight = Total(Temporary(weights_use),3)
 i_use = Where(weight,n_use)
 vis_xy = Reform(vis_xy[i_use],1,n_use)
 vis_yx = Reform(vis_yx[i_use],1,n_use)
+model_xy = Reform(model_xy[i_use],1,n_use)
+model_yx = Reform(model_yx[i_use],1,n_use)
 
-vis_sum = Total(Temporary(vis_xy)*Conj(Temporary(vis_yx)))
-cross_phase = 0.5*atan(vis_sum, /phase)
+vis_sum = Total(Conj(Temporary(vis_xy))*Temporary(model_xy)) + Total(Conj(Temporary(vis_yx))*Temporary(model_yx))
+cross_phase = atan(vis_sum, /phase)
+
+print,"Phase fit between X and Y antenna polarizations:", cross_phase
 
 cal.cross_phase = cross_phase
-*(cal.gain[0]) *= Exp(-icomp * cross_phase / 2.0)
-*(cal.gain[1]) *= Exp(icomp * cross_phase / 2.0)
+*(cal.gain[0]) *= Exp(-icomp*cross_phase/2.0)
+*(cal.gain[1]) *= Exp(icomp*cross_phase/2.0)
+
+*vis_ptr[2] *= Exp(icomp*cross_phase)
+*vis_ptr[3] *= Exp(-icomp*cross_phase)
```

Comment:
I don't think I like modifying the visibilities outside of `vis_calibration_apply`. Since we store `cal.cross_phase`, I think the correct way would be to not modify the `cal.gain` or `vis_ptr` here at all, but include new code in `vis_calibration_apply` along the lines of:
```
IF n_pol EQ 4 THEN BEGIN
*vis_ptr[2] *= Exp(icomp*cal.cross_phase)
*vis_ptr[3] *= Exp(-icomp*cal.cross_phase)
ENDIF
```

Reply:
That makes complete sense. The only problem is that in this implementation vis_calibrate_crosspol_phase uses the calibrated visibilities (calibrated up to the crosspol phase), not the raw visibilities. How would you recommend doing this?

Reply:
I'm going to push a commit with my suggested changes. Please look those over and verify that those do what you want.

Reply:
I've fixed the bugs in those changes. The new code has been tested and returns the expected result. I'm going to go ahead and merge.